app/Http/Controllers/Auth/ForgotPasswordController.php

```diff
@@ -43,30 +47,36 @@ public function sendResetLinkEmail(Request $request)
         $this->validateEmail($request);
 
         $email_address = $request->get('email');
-        $token_alias = hash('sha256', $email_address.date("D M d, Y G:i"));
+        $token_alias = $this->generateToken($email_address);
 
-        PasswordReset::where('email', 'like', $email_address)->delete(); // Make sure a token for the specified email is non-existent
+        // Make sure a token for the specified email is non-existent.
+        // This ensures that the specified email address will not have
+        // more than one (1) tokens.
+        PasswordReset::where('email', 'like', $email_address)->delete();
         PasswordReset::create([
             'email' => $email_address,
             'token' => PasswordReset::generateToken($token_alias),
         ]);
 
-        $email = new \SendGrid\Mail\Mail(); 
-        $email->setFrom("noreply@knowlaravel.com", "Password Reset");
-        $email->setSubject("Reset your password");
-        $email->addTo($request->get('email'));
-        $email->addContent(
-            "text/html", "<a href='".route('password.reset', ['token' => $token_alias])."'>Click here to reset your password</a>"
-        );
-        
-        $sendgrid = new \SendGrid(getenv('SENDGRID_API_KEY'));
-        try {
-            $response = $sendgrid->send($email);
-            $status = "You will receive a reset link if the email address you provided is correct.";
-        } catch (Exception $e) {
-            $status = $e->getMessage();
-        }
+        $mailer = Mailer::create(Mailer::SENDGRID);
```

Comment:
Make the email driver configurable. To easily switch driver without changing the code.

Put on the settings table default is SendGrid